make bounce handling somewhat more robust

justcfx2u opened the pull request:
1. Set `$mail->Host` to avoid:

    ```production.ERROR: ErrorException: stream_socket_enable_crypto(): Peer certificate CN=`fully.qualified.domain' did not match expected CN=`localhost' in /opt/abuseio/vendor/kruisdraad/phpmailer/class.smtp.php:368```

2. Preserve all "To" addresses.

~3. Use `getAddresses('to')` because PHPMailer does not like the output of `"display name" <user@example.com>' provided by php-mail-mime-parser's getHeader('to') method.~

~4. Use `getAddresses('from')` because PHPMailer does not like the output of `"display name" <user@example.com>' provided by php-mail-mime-parser's getHeader('from') method.~

~5. Upgrade php-mime-mail-parser dependency to v2.6.0 or better. The getAddresses() method was added at this revision.~

kruisdraad commented:
point 1, great idea added some more related options

point 2,3,4, needs a 2.1.6 parser implementation, see 5

point 5, upgrading from 2.1.6 implements major breakage. You'd end up with only be able to bounce and all the rest of mail handling will be fully broken. We would look into a major overhaul into 2.6.0 with the abuseio 4.1 release but that needs some investigation

justcfx2u commented:
I have addressed points 2-5 by reverting composer.json, and am now parsing out the raw addresses from the getHeader output, which PHPMailer requires.

The code now works nicely in my limited testing.

justcfx2u commented:
consolidated changes with git reset/rebase

kruisdraad commented:
Code is correctly formatted and passes checks

thank you for your work on our project :)
